src/Adapter/Order/Refund/OrderSlipCreator.php

```diff
@@ -138,13 +147,184 @@ public function createOrderSlip(
             );
 
             /** @var OrderDetail $orderDetail */
-            foreach ($orderRefundDetail->getOrderDetails() as $orderDetail) {
+            foreach ($orderRefundSummary->getOrderDetails() as $orderDetail) {
                 if ($this->configuration->get('PS_ADVANCED_STOCK_MANAGEMENT')) {
                     StockAvailable::synchronize($orderDetail->product_id);
                 }
             }
         } else {
-            throw new EmptyRefundAmountException();
+            throw new InvalidRefundAmountException();
         }
     }
+
+    /**
+     * This is a copy of OrderSlip::create except the OrderDetail modification has been removed
+     * since it's now managed in the handler, this allows to update order details even without
+     * generating a credit slip
+     *
+     * @todo this copy uses array data but could probably be refactored to use OrderDetailRefund objects
+     *
+     * @param Order $order
+     * @param array $product_list
+     * @param float $shipping_cost
+     * @param float $amount
+     * @param bool $amount_choosen
+     * @param bool $add_tax
+     * @param int $precision
+     *
+     * @return bool
+     *
+     * @throws PrestaShopDatabaseException
+     * @throws PrestaShopException
+     */
+    private function createOrderSlip(Order $order, array $product_list, float $shipping_cost = 0, float $amount = 0, bool $amount_choosen = false, bool $add_tax = true, int $precision = 6)
+    {
+        $currency = new Currency((int) $order->id_currency);
+        $order_slip = new OrderSlip();
```

Comment:
```suggestion
        $orderSlip = new OrderSlip();
```
You can use camelCase :)

Reply:
Yes it was copy/pasted from legacy code but I will clean it a bit ;)

Reply:
Done